sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1005,7 +1006,8 @@ private[hive] object HiveClientImpl extends Logging {
   /** Builds the native StructField from Hive's FieldSchema. */
   def fromHiveColumn(hc: FieldSchema): StructField = {
     val columnType = getSparkSQLDataType(hc)
-    val metadata = if (hc.getType != columnType.catalogString) {
+    val metadata = if (hc.getType != columnType.catalogString &&
+        hc.getType != HiveVoidType.catalogString) {
```

Comment:
why do we need this?

Reply:
Avoid void type add to Metadata.

Without this, Hive `void` type will be put in Metadata when we load hive column.

Reply:
what's the drawback of it?

BTW this check only works for top-level void columns, not inner fields.

Reply:
No sure it will be, just think we do not need to use void type in other place.